Library/Formula/libbpg.rb

```diff
@@ -11,24 +12,38 @@ class Libbpg < Formula
     sha256 "7b8d1585ee9e2de010abfab1aaab419d052cf35d82704587c2f75947d4fc8ee5" => :mavericks
   end
 
-  option "with-x265", "Enable x265 encoder"
-  option "without-jctvc", "Disable built-in JCTVC encoder"
+  option "with-jctvc", "Enable built-in JCTVC encoder - Mono threaded, slower but produce smaller file"
+  option "without-x265", "Disable built-in x265 encoder - Multi threaded, faster but produce bigger file"
 
   depends_on "cmake" => :build
-  depends_on "yasm" => :build
+  depends_on "yasm" => :build if build.with? "x265"
   depends_on "libpng"
   depends_on "jpeg"
-  depends_on "x265" => :optional
 
   def install
     bin.mkpath
-    args = []
-    args << "USE_X265=y" if build.with? "x265"
-    args << "USE_JCTVC=" if build.without? "jctvc"
-    system "make", "install", "prefix=#{prefix}", "CONFIG_APPLE=y", *args
+
+    makeArgs = []
+    makeArgs << "install"
+    makeArgs << "prefix=#{prefix}"
+    makeArgs << "CONFIG_APPLE=y"
+    makeArgs << "USE_JCTVC=y" if build.with? "jctvc"
+    makeArgs << "USE_X265=" if build.without? "x265"
+
+    system "make", *makeArgs
   end
 
   test do
-    system "#{bin}/bpgenc", test_fixtures("test.png")
+    if build.with? "jctvc"
+      system "#{bin}/bpgenc", "-e", "jctvc", test_fixtures("test.png")
+      # Unable to test 1x1 jpg
+      #system "#{bin}/bpgenc", "-e", "jctvc", test_fixtures("test.jpg")
+    end
+
+    if build.with? "x265"
+      system "#{bin}/bpgenc", "-e", "x265", test_fixtures("test.png")
+      # Unable to test 1x1 jpg
+      #system "#{bin}/bpgenc", "-e", "x265", test_fixtures("test.jpg")
+    end
```

Comment:
In this case, if you want only old (and best) encoder `--with-jctvc --without-x265`, test will always fail. If it's ok for you, I'll remove these cases.